src/components/Login/index.jsx

```diff
@@ -1,119 +1,137 @@
-      isShowingError: false,
-      userName: "",
-      password: ""
-    };
-
-    this.handleLogin = this.handleLogin.bind(this);
-  }
-
-  updateField(e, field) {
-    this.setState({
-      [field]: e.target.value
-    });
-  }
-
-  handleLogin() {
-    const { userName, password } = this.state;
-
-    if (userName === "ziv" && password === "papa") {
-      this.setState(prevState => ({
-        isAuthenticated: !prevState.isAuthenticated
-      }));
-    } else {
-      this.setState(prevState => ({
-        isShowingError: !prevState.isShowingError
-      }));
-    }
-  }
-
-  render() {
-    const { isAuthenticated, isShowingError } = this.state;
-    const redirectLink = "/bounce_rules";
-
-    return isAuthenticated ? (
-      <Redirect to={redirectLink} />
-    ) : (
-      <div
-        className="row"
-        style={{
-          display: "flex",
-          justifyContent: "center",
-          alignItems: "center",
-          height: "100vh"
-        }}
-      >
-        <div
-          style={{
-            display: "flex",
-            justifyContent: "center"
-          }}
-        >
-          <div
-            className="card is-centered"
-            style={{
-              width: "30rem",
-              height: "30rem",
-              display: "flex",
-              flexDirection: "column",
-              justifyContent: "center",
-              alignItems: "center"
-            }}
-          >
-            <div
-              className="login-title"
-              style={{
-                fontWeight: "bold"
-              }}
-            >
-              SendGrid
-            </div>
-            {isShowingError && (
-              <div style={{ color: "#b71c1c", marginTop: "1rem" }}>
-                Incorrect Credentials
-              </div>
-            )}
-            <div
-              className="login-form"
-              style={{
-                marginBottom: "2rem",
-                width: "20rem"
-              }}
-            >
-              <div className="input-text-wrap">
-                <input
-                  type="text"
-                  placeholder="Username"
-                  onChange={e => this.updateField(e, "userName")}
+const Login = ({
+  username,
+  password,
+  isAuthenticating,
+  isAuthenticationError,
+  isInvalidCredentials,
+  isInvalidInput,
+  isNetworkError,
+  handleLoginSubmit,
+  handleInputChange,
+  handleAlertClose,
+}) => (
+  <div className="login-container">
+    <Row>
+      <Column width={10} offset={2}>
+        <div className="login-form-container">
+          <div className="login-logo">
+            <img
+              src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
+              alt="sg-logo"
+            />
+          </div>
+          <Row width={8} offset={2}>
+            {!isAuthenticating &&
+              isAuthenticationError &&
+              isInvalidCredentials && (
+                <InvalidCredentialsAlert
```

Comment:
That's awesome you guys broke it into SFCs for the alerts to be more readable. Since these alerts are very domain specific to the login page we don't need to break it into its own file and folder as they probably won't be shared. When it comes to components like these, it's better to leave them in the same file as this so you don't have to open many different folders/files to reach the SFCs that will most likely only be used in this page.
Above this you can declare all the alerts and the login button and loading button like
```js
const InvalidCredentialsAlert = ({ }) => (
...);
const NetworkErrorAlert = ({ }) => (
...
);
```